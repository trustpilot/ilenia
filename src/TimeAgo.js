import withTranslations from './withTranslations';
import PropTypes from 'prop-types';
import timeago from 'timeago.js';

import da from 'timeago.js/locales/da';
import de from 'timeago.js/locales/de';
import en from 'timeago.js/locales/en';
import es from 'timeago.js/locales/es';
import fr from 'timeago.js/locales/fr';
import it from 'timeago.js/locales/it';
import nl from 'timeago.js/locales/nl';
import sv from 'timeago.js/locales/sv';

timeago.register('da-DK', da);
timeago.register('de-DE', de);
timeago.register('en-GB', en);
timeago.register('en-US', en);
timeago.register('es-ES', es);
timeago.register('fr-FR', fr);
timeago.register('it-IT', it);
timeago.register('nl-NL', nl);
timeago.register('sv-SE', sv);

const TimeAgo = ({ date, locale }) => {
  const dateWrapper = new Date(date);
  const timeagoInstance = timeago();

  if (isNaN(dateWrapper)) {
    // eslint-disable-next-line no-console
    console.error('Invalid date');
    return null;
  }

  return timeagoInstance.format(date, locale);
};

TimeAgo.propTypes = {
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.instanceOf(Date)]),
};

export default withTranslations(TimeAgo);
