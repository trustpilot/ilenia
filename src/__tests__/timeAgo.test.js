import React from 'react';
import renderer from 'react-test-renderer';

import LocalizationProvider from '../LocalizationProvider';
import TimeAgo from '../TimeAgo';

test('Renders a localized date from a number', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <TimeAgo date={Date.now() - 11 * 1000 * 60 * 60} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized date from a string', () => {
  const component = renderer.create(
    <LocalizationProvider locale="es-ES" translations={{}}>
      <TimeAgo date={new Date(Date.now() - 11 * 1000 * 60 * 60).toString()} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized date from a date', () => {
  const component = renderer.create(
    <LocalizationProvider locale="fr-FR" translations={{}}>
      <TimeAgo date={new Date(Date.now() - 11 * 1000 * 60 * 60)} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Throw an error if the date is invalid', () => {
  //eslint-disable-next-line no-console
  console.error = jest.fn();

  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <TimeAgo date={'abcdefghijklmnopqrstuvwxyz'} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  //eslint-disable-next-line no-console
  expect(console.error).toHaveBeenCalled();
});

test('Renders a localized date with an incorrect locale', () => {
  const component = renderer.create(
    <LocalizationProvider locale="abcdefghijklmnopqrstuvwxyz" translations={{}}>
      <TimeAgo date={new Date(Date.now() - 11 * 1000 * 60 * 60)} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
