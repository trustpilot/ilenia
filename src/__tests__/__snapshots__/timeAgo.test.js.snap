// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a localized date from a date 1`] = `"il y a 11 heures"`;

exports[`Renders a localized date from a number 1`] = `"11 hours ago"`;

exports[`Renders a localized date from a string 1`] = `"hace 11 horas"`;

exports[`Renders a localized date with an incorrect locale 1`] = `"11 hours ago"`;

exports[`Throw an error if the date is invalid 1`] = `null`;
